backend/app/controllers/api/v3/jobs_controller.rb

```diff
@@ -0,0 +1,718 @@
+      # @param token [String] a token
+      def unassign
+        # Check for manage permissions
+        status, response = check_token_for_permission(Permission.manage_id)
+        render json: response.to_json, status: status.to_sym and return if response[:error]
+
+        # Get job
+        @id = Input.int(params[:id])
+        job = Job.find_by(id: @id)
+        render json: { error: "Job not found" }.to_json, status: :not_found and return if !job
+
+        @by = response[:user]['id'].to_i
+        @to = nil
+
+        # Create assignment
+        new_job_assignment_log
+      end
+
+      # Creates a job
+      #
+      # <b>API Call:</b>
+      #   GET: /api/v3/jobs/create
+      #   {
+      #     token: <token>,
+      #     operation_ids[]: <operation_id>,
+      #     operation_ids[]: <operation_id>,
+      #     ...
+      #   }
+      #
+      # <b>API Return Success:</b>
+      #   STATUS_CODE: 200
+      #   {
+      #     job: {
+      #       id: <___>,
+      #       user_id: <___>,
+      #       arguments: null,
+      #       state: [
+      #         {
+      #           operation: "initialize",
+      #           arguments: {
+      #             operation_type_id: <operation_type_id>,
+      #             time: <timenow>
+      #           }
+      #         }
+      #       ]
+      #       created_at: <created_at>,
+      #       updated_at: <updated_at>,
+      #       path: "operation.rb",
+      #       pc: -1,
+      #       group_id: null,
+      #       submitted_by: <submitted_by>,
+      #       desired_start_time: <timenow>,
+      #       latest_start_time: <timenow + 1.hour>,
+      #       metacol_id: null,
+      #       successor_id: null
+      #     }
+      #   }
+      # @!method create(token, operation_ids[])
+      # @param token [String] a token
+      # @param operation_ids [Array] the list of operation_ids
+      def create
+        # Check for manage permissions
+        status, response = check_token_for_permission(Permission.manage_id)
+        render json: response.to_json, status: status.to_sym and return if response[:error]
+
+        # Read operation ids
+        operation_ids = [0]
+        params[:operation_ids].each do |operation_id|
+          operation_ids << operation_id
+        end
+
+        # Get pending operations
+        operations = Operation.pending_operations(operation_ids)
+        render json: { error: "No pending operations selected" }.to_json, status: :unauthorized and return if operations.length == 0
+
+        # Check that operation_type_ids are the same
+        distinct = Operation.distinct_operation_types(operation_ids, 'pending')
+        render json: { error: "Cannot combine operations with different operation types" }.to_json, status: :unauthorized and return if distinct.length > 1
+
+        # Set the state for the job
+        # Set timenow so that the time matches created_at and updated_at for the job
+        timenow = Time.now.utc
```

Comment:
I do this because there are multiple timestamps that should be the same (Rails would normally set "created_at" and "updated_at" automatically. But the database also creates a log  entry (as json) in the "state" field, sets a "desired_start_time" and sets a "latest_start_time".

Until we change the actual logic of the system we should at least keep the current architecture "clean".  The best way to do that is to capture the time as a variable and re-use it.